docs/style-guide.md

```diff
@@ -0,0 +1,612 @@
+    this.newPhoneButton.click();
+    $$('#phone-list .phone-row').first().then(function(row) {
+      row.element(by.model('item.phoneNumber')).sendKeys(phone);
+      row.element(by.model('item.contactType')).sendKeys(contactType);
+    });
+  };
+};
+```
+
+**Why?**
+* Most elements are already exposed by the page object and can be used
+  directly in the test.
+* Doing otherwise will not have any added value
+
+### Avoid using expect() in page objects
+
+* Don't make any assertions in your page objects.
+
+**Why?**
+* It is the responsibility of the test to do all the assertions.
+* A reader of the test should be able to understand the behavior of the
+  application by looking at the test only
+
+### Add page object wrappers for directives, dialogs, and common elements
+
+* Some directives render complex HTML or they change frequently. Avoid code
+  duplication by writing wrappers to interact with complex directives.
+* Dialogs or modals are frequently used across multiple views.
+* When the directive changes you only need to change the wrapper once.
+
+For example, the Protractor website has navigation bar with multiple dropdown
+menus. Each menu has multiple options. A page object for the menu would look
+like this:
+
+```js
+/**
+ * Page object for Protractor website menu.
+ * @constructor
+ */
+var MenuPage = function() {
+  this.dropdown = function(dropdownName) {
+    /**
+     * Dropdown api. Used to click on an element under a dropdown.
+     * @param {string} dropdownName
+     * @return {{option: Function}}
+     */
+    var openDropdown = function() {
+      element(by.css('.navbar-nav'))
+          .element(by.linkText(dropdownName))
+          .click();
+    };
+
+    return {
+      /**
+       * Get an option element under a dropdown.
+       * @param {string} optionName
+       * @return {ElementFinder}
+       */
+      option: function(optionName) {
+        openDropdown();
+        return element(by.css('.dropdown.open'))
+            .element(by.linkText(optionName));
+      }
+    }
+  };
+};
+
+module.exports = MenuPage;
+```
+
+```js
+var Menu = require('./menu');
+
+describe('protractor website', function() {
+
+  var menu = new Menu();
+
+  it('should navigate to API view', function() {
+    browser.get('http://www.protractortest.org/#/');
+
+    menu.dropdown('Reference').option('Protractor API').click();
+
+    expect(browser.getCurrentUrl())
+        .toBe('http://www.protractortest.org/#/api');
+  });
+});
+```
+
+**Why?**
+* When you have a large team and multiple e2e tests people tend to write
+  their own custom locators for the same directives.
+
+### Avoid text locators for text that changes frequently
+
+* Try to avoid text-based locators such as `by.linkText`, `by.buttonText`,
+  `by.cssContainingText`.
+
+**Why?**
+* Text for buttons, links, and labels tends to change over time. Minor text
+  changes in your application should not break your tests.
+
+# Test suites
```

Comment:
I think this section has some of the most important advice to avoid the most common bad tests. Maybe we can move the whole section up to emphasize its importance? What do you think?